.github/workflows/validate_pyinstaller.yml

```diff
@@ -39,7 +39,7 @@ jobs:
 
   build-for-mac:
     name: build-pyinstaller-macos
-    runs-on: macos-13
```

Comment:
I think this was pinned to 13 before since that's what our installer system uses. Can we revert this back?

Reply:
Why would we not want to keep it pinned to the version that most closely matches our internal systems?

Reply:
After giving it some more thought, I don't think it matters much which way we go here. Let's get these merged in